### PR TITLE
planner: merge userVar and userVarType

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by fzzf678** [comment]
> Please follow PR Title Format:
> 
> * pkg [, pkg2, pkg3]: what is changed
> 
> Or if the count of mainly changed packages are more than 3, use
> 
> * *: what is changed
> 
> After you have format title, you can leave a comment `/run-check_title` to recheck it

/mute

File: expression/builtin_other.go [hunk 1]
```diff
@@ -1027,8 +1046,10 @@ func (b *builtinGetIntVarSig) evalInt(row chunk.Row) (int64, bool, error) {
 	varName = strings.ToLower(varName)
 	sessionVars.UsersLock.RLock()
 	defer sessionVars.UsersLock.RUnlock()
-	if v, ok := sessionVars.Users[varName]; ok {
-		return v.GetInt64(), false, nil
+	if useVar, ok := sessionVars.UserVars.Vars[varName]; ok {
+		if v, ok1 := useVar.(*Constant); ok1 {
+			return v.Value.GetInt64(), false, nil
```
Comment (qw4990) [inline expression/builtin_other.go:1051]: We can use `Constant.EvalInt` directly here, `return var.(*Constant).EvalInt(ctx, chunk.Row{})`

File: planner/core/plan_cache.go [hunk 2]
```diff
@@ -102,8 +102,11 @@ func parseParamTypes(sctx sessionctx.Context, params []expression.Expression) (p
 
 		// from text protocol, there must be a GetVar function
 		name := param.(*expression.ScalarFunction).GetArgs()[0].String()
-		tp := sctx.GetSessionVars().UserVarTypes[name]
-		if tp == nil {
+		userVar, ok := sctx.GetSessionVars().UserVars.Vars[name]
+		var tp *types.FieldType
+		if v, ok1 := userVar.(*expression.Constant); ok && ok1 {
+			tp = v.RetType
```
Comment (qw4990) [inline planner/core/plan_cache.go:108]:
```suggestion
tp = v.GetType()
```

**Pull request closed without merging**